File: golang/kurtosis-lambda/impl/example_kurtosis_lambda_configurator.go
package impl

import (
	"encoding/json"
	kurtosis_lambda "github.com/kurtosis-tech/kurtosis-lambda-api-lib/golang/lib/kurtosis-lambda"
	"github.com/palantir/stacktrace"
	"github.com/sirupsen/logrus"
)

const(
	defaultLogLevel = "info"
)

type ExampleKurtosisLambdaConfigurator struct{}

func NewExampleKurtosisLambdaConfigurator() *ExampleKurtosisLambdaConfigurator {
	return &ExampleKurtosisLambdaConfigurator{}
}

func (t ExampleKurtosisLambdaConfigurator) ParseParamsAndCreateKurtosisLambda(serializedCustomParamsStr string) (kurtosis_lambda.KurtosisLambda, error) {
	serializedCustomParamsBytes := []byte(serializedCustomParamsStr)
	var args ExampleKurtosisLambdaArgs
	if err := json.Unmarshal(serializedCustomParamsBytes, &args); err != nil {
		return nil, stacktrace.Propagate(err, "An error occurred deserializing the Kurtosis Lambda serialized custom params with value '%v", serializedCustomParamsStr)
	}

	err := setLogLevel(args.LogLevel)
	if err != nil {
		return nil, stacktrace.Propagate(err, "An error occurred setting the log level")
	}

	lambda := NewExampleKurtosisLambda()

	return lambda, nil
}

func setLogLevel(logLevelStr string) error {
	if logLevelStr == "" {
		logLevelStr = defaultLogLevel
	}
	level, err := logrus.ParseLevel(logLevelStr)
	if err != nil {
		return stacktrace.Propagate(err, "An error occurred parsing loglevel string '%v'", logLevelStr)
	}

	logrus.SetLevel(level)
	logrus.SetFormatter(&logrus.TextFormatter{
		ForceColors:   true,
		FullTimestamp: true,
	})
	return nil
}

File: typescript/kurtosis-lambda/impl/example_kurtosis_lambda_configurator.ts
import { KurtosisLambda, KurtosisLambdaConfigurator } from 'kurtosis-lambda-api-lib';
import { Result, err, ok } from 'neverthrow';
import * as log from 'loglevel';

const DEFAULT_LOG_LEVEL: string = "info";

type LoglevelAcceptableLevelStrs = log.LogLevelDesc

class ExampleKurtosisLambdaConfigurator implements KurtosisLambdaConfigurator {
	public parseParamsAndCreateKurtosisLambda(serializedCustomParamsStr: string): Result<KurtosisLambda, Error> {
		let args: ExampleKurtosisLambdaArgs;
		try {
			args = JSON.parse(serializedCustomParamsStr);
		} catch (e: Error) {
			return err(e)
		}

		const setLogLevelResult: Result<null, Error> = ExampleKurtosisLambdaConfigurator.setLogLevel(args.getLogLevel())
		if (setLogLevelResult.isErr()) {
			return err(setLogLevelResult.error);
		}

		const lambda: KurtosisLambda = new ExampleKurtosisLambdaConfigurator();
		return ok(lambda);
	}

	private static setLogLevel(logLevelStr: string): Result<null, Error> {
		let logLevelDescStr: string = logLevelStr;
		if (logLevelStr === null || logLevelStr === undefined || logLevelStr === "") {
			logLevelDescStr = DEFAULT_LOG_LEVEL;
		}
		const logLevelDesc: log.LogLevelDesc = logLevelDescStr as log.LogLevelDesc
		log.setLevel(logLevelDesc);
		return ok(null);
	}
}